x-pack/plugins/watcher/public/sections/watch_edit/components/threshold_watch_edit_component.tsx

```diff
@@ -421,34 +439,39 @@ const ThresholdWatchEditUi = ({
                   <EuiPopoverTitle>of</EuiPopoverTitle>
                   <EuiFlexGroup>
                     <EuiFlexItem grow={false} style={{ width: 150 }}>
-                      <EuiSelect
-                        value={watch.aggField}
-                        onChange={e => {
-                          setWatchProperty('aggField', e.target.value);
-                          setAggFieldPopoverOpen(false);
-                        }}
-                        options={fields.reduce(
-                          (options, field: any) => {
-                            if (
-                              aggTypes[watch.aggType].validNormalizedTypes.includes(
-                                field.normalizedType
-                              )
-                            ) {
-                              options.push({
-                                text: field.name,
-                                value: field.name,
-                              });
+                      <ErrableFormRow
+                        errorKey="aggField"
+                        isShowingErrors={hasErrors && watch.aggField !== undefined}
+                        errors={errors}
+                      >
+                        <EuiSelect
+                          value={watch.aggField}
+                          onChange={e => {
+                            setWatchProperty('aggField', e.target.value);
+                          }}
+                          onBlur={() => {
+                            if (!watch.aggField) {
+                              setWatchProperty('aggField', '');
                             }
-                            return options;
-                          },
-                          [
-                            {
-                              text: 'select a field',
-                              value: '',
+                          }}
+                          options={fields.reduce(
```

Comment:
getting a few react warnings related to this i think

```React does not recognize the `fieldRequired` prop on a DOM element```
```React does not recognize the `validNormalizedTypes` prop on a DOM element.```

Reply:
found and fixed this